baseapp/abci.go

```diff
@@ -66,9 +67,26 @@ func (app *BaseApp) InitChain(req abci.RequestInitChain) (res abci.ResponseInitC
 		}
 	}
 
+	// In the case of a new chain, AppHash will be the hash of an empty string.
+	// During an upgrade, it'll be the hash of the last committed block.
+	var appHash []byte
+	if !app.LastCommitID().IsZero() {
+		appHash = app.LastCommitID().Hash
```

Comment:
> Added line #L74 was not covered by tests

Is there a test somewhere with InitChain called after an upgrade, from which I can take inspiration?

Reply:
there's none afaik. May be we can try something with `cosmovisor`